### PR TITLE
Migrate some ipc container test from integration-cli to integration

### PR DESCRIPTION
**Pull request by yongtang**
This fix migrates some ipc container tests from integration-cli to integration test.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

File: integration/container/ipcmode_test.go [hunk 1]
```diff
@@ -114,3 +114,70 @@ func TestIpcModeShareable(t *testing.T) {
...
+
+// TestAPIIpcModeShareableAndContainer checks that a container created with
+// --ipc container:ID can use IPC of another shareable container.
+func TestAPIIpcModeShareableAndContainer(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType != "linux")
+
+	testIpcContainer(t, "shareable", true)
+}
+
+// TestAPIIpcModePrivateAndContainer checks that a container created with
```
Comment (thaJeztah) [inline integration/container/ipcmode_test.go]: Whoops, looks like you forgot to copy the second line of this test's

File: integration/container/ipcmode_test.go [hunk 2]
```diff
@@ -114,3 +114,70 @@ func TestIpcModeShareable(t *testing.T) {
 
 	testIpcNonePrivateShareable(t, "shareable", true, true)
 }
+
+// testIpcContainer is a helper function to test --ipc container:NNN mode in various scenarios
+func testIpcContainer(t *testing.T, donorMode string, mustWork bool) {
+	defer setupTest(t)()
```
Comment (thaJeztah) [inline integration/container/ipcmode_test.go]: Should this function have `t.Helper()`?

File: integration/container/ipcmode_test.go [hunk 3]
```diff
@@ -114,3 +114,70 @@ func TestIpcModeShareable(t *testing.T) {
...
+	out := result.Combined()
+	assert.Check(t, is.Equal(true, regexp.MustCompile("^covfefe$").MatchString(out)))
+}
+
+// TestAPIIpcModeShareableAndContainer checks that a container created with
+// --ipc container:ID can use IPC of another shareable container.
+func TestAPIIpcModeShareableAndContainer(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType != "linux")
+
+	testIpcContainer(t, "shareable", true)
```
Comment (thaJeztah) [inline integration/container/ipcmode_test.go]: (just a suggestion); we could combine these two tests and use subtests (one with `shareable` one with `private`)

**Review by thaJeztah — CHANGES_REQUESTED** [review]
thanks! some minor comments / suggestions

**Comment by yongtang** [comment]
@thaJeztah Thanks. The PR has been updated. Please take a look.

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Pull request merged**